diff --git a/js/lightbox.js b/js/lightbox.js
index 0f7213f..ae28693 100644
--- a/js/lightbox.js
+++ b/js/lightbox.js
@@ -1072,8 +1072,9 @@ Lightbox = {
         parts["flashvars"] = parts["style"].match(/flashvars:\s?(.*?);/)[1];
       }
     }
-    if ($(link).attr('rel').match(/\[.*\]\[(.*)\]/)) {
-      parts["title"] = $(link).attr('rel').match(/\[.*\]\[(.*)\]/)[1];
+    if ($(link).attr('rel').match(/\[.*\]\[([\s\S]*)\]/)) {
+      var title = $(link).attr('rel').match(/\[.*\]\[([\s\S]*)\]/)[1];
+      parts["title"] = title.replace(/\n/g, '<br />');
     }
     return parts;
   },
